R/moma_arguments.R

```diff
@@ -1,80 +1,83 @@
 #' penalty functions.
 #' \itemize{
-#'   \item{\code{\link{lasso}}} TODO
-#'   \item{\code{\link{mcp}}} TODO
-#'   \item{\code{\link{scad}}} TODO
-#'   \item{\code{\link{slope}}} TODO
-#'   \item{\code{\link{grplasso}}} TODO
-#'   \item{\code{\link{fusedlasso}}} TODO
-#'   \item{\code{\link{l1tf}}} TODO
-#'   \item{\code{\link{spfusedlasso}}} TODO
-#'   \item{\code{\link{cluster}}} TODO
+#'   \item{\code{\link{moma_lasso}}}: sparsity
+#'   \item{\code{\link{moma_mcp}}}: adaptive sparsity
+#'   \item{\code{\link{moma_scad}}}: adaptive sparsity
+#'   \item{\code{\link{moma_slope}}}: adaptive sparsity
+#'   \item{\code{\link{moma_grplasso}}}: group-wise sparsity
+#'   \item{\code{\link{moma_fusedlasso}}}: piece-wise constant
+#'   \item{\code{\link{moma_spfusedlasso}}}: sparsity and piece-wise constant
+#'   \item{\code{\link{moma_l1tf}}}: piece-wise quadratic
+#'   \item{\code{\link{moma_cluster}}}: structured sparsity
 #' }
-#' @name moma_sparsity
+#' These functions specify the value of the \code{u_sparse,v_sparse} arguments in the
+#'  \code{moma_*pca} series of functions, and the \code{x_sparse,y_sparse} arguments
+#' in the \code{moma_*cca} and \code{moma_*lda} series of functions.
+#'
+#' All functions
+#' above share two common parameters: \code{lambda} and \code{select_scheme}, which are
+#' described in the Arguments section below.
+#' @name moma_sparsity_options
+#' @param ... Force users to specify arguments by names.
+#' @param lambda A vector containing penalty values
+#' @param select_scheme A char being either "b" (nested BIC search) or "g" (grid search).
+#'
+#' Selection of tuning parameters (four parameters in our case, \code{alpha_u}, \code{alpha_v}, \code{lambda_u}, \code{lambda_v}) is a problem in MoMA. Currently we provide
+#' two modes: grid search ("g") and nested BIC search ("b"). Grid search means we solve the problem
+#' for all combinations of parameter values provided by the user.
+#'
+#' To explain nested BIC search, we need to look into how the algorithm runs.
+#' To find an (approximate) solution to a penalized SVD (Singular Value Decomposition) problem is to solve two
+#' penalized regression problems iteratively. Let's call them problem u and problem v, which give
+#' improving estimates of the right singular vector, \emph{u}, and the left singular vector, \emph{v}, respectively.
+#' For each regression problem, we can select the optimal parameters
+#' based on BIC.
+#'
+#' The nested BIC search is essentially two 2-D searches. We start from SVD solutions, and then find the optimal
+#' parameters for problem u, given current estimate of \emph{v}. Using the result from previous step, update
+#' current estimate of \emph{u}, and then do the same thing for problem v,
+#' that is, to find the optimal parameters for problem v given current estimate of \emph{u}. Repeat
+#' the above until convergence or the maximal number of iterations has been reached.
```

Comment:
You can also point folks to the brief discussion in the paper. 

This definitely is the sort of thing we should write a vignette on at some post-GSoC point.

Reply:
Reference added.

Yeah, for those not familiar with the paper it would be hard to understand the selection procedure. The "functional data analysis" vignetted demonstrates the use of `select_scheme = "b"`. 

Opened an issue to track this #57.